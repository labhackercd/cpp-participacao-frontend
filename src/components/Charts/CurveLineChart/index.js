import React from 'react';
import Chart from "react-google-charts";


export default function CurveLineChart(props) {
  return (
    <Chart
      width={'500px'}
      height={'300px'}
      chartType="LineChart"
      loader={<div>Carregando Gr√°fico</div>}
      data={props.data}
      options={props.options}
      rootProps={{ 'data-testid': '2' }}
    />
   );
}
