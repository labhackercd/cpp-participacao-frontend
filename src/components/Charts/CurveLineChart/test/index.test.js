import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from "enzyme";
import CurveLineChart from './../index';

const curveLineChartdata = {
    values:[
        ['ano', 'UsuÃ¡rios'],
        ['2016', 134],
        ['2017', 523],
        ['2018', 400],
        ['2019', 843],
        ['2020', 1432],
      ],
    options:{
        backgroundColor: '#000000',
        hAxis: {
          textStyle:{color: '#FFFFFF'}
        },
        vAxis: {
          textPosition: 'none',
          gridlines: {color: 'transparent'}
        },
        curveType: 'function',
        legend:{textStyle: {color: 'white'}},
      }
};

test("Snapshot should not have changes", () => {
    const component = mount(<CurveLineChart data={curveLineChartdata.values} options={curveLineChartdata.options}></CurveLineChart>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

it("Mounts the chart with the data", () => {
    const div = document.createElement("div")
    ReactDOM.render(<CurveLineChart data={curveLineChartdata.values} options={curveLineChartdata.options}></CurveLineChart>, div)

});
