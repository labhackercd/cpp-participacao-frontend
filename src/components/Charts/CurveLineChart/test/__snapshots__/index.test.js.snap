// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should not have changes 1`] = `
<CurveLineChart
  data={
    Array [
      Array [
        "ano",
        "Usuários",
      ],
      Array [
        "2016",
        134,
      ],
      Array [
        "2017",
        523,
      ],
      Array [
        "2018",
        400,
      ],
      Array [
        "2019",
        843,
      ],
      Array [
        "2020",
        1432,
      ],
    ]
  }
  options={
    Object {
      "backgroundColor": "#000000",
      "curveType": "function",
      "hAxis": Object {
        "textStyle": Object {
          "color": "#FFFFFF",
        },
      },
      "legend": Object {
        "textStyle": Object {
          "color": "white",
        },
      },
      "vAxis": Object {
        "gridlines": Object {
          "color": "transparent",
        },
        "textPosition": "none",
      },
    }
  }
>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="LineChart"
    chartWrapperParams={Object {}}
    className=""
    columns={null}
    controls={null}
    data={
      Array [
        Array [
          "ano",
          "Usuários",
        ],
        Array [
          "2016",
          134,
        ],
        Array [
          "2017",
          523,
        ],
        Array [
          "2018",
          400,
        ],
        Array [
          "2019",
          843,
        ],
        Array [
          "2020",
          1432,
        ],
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="300px"
    legendToggle={false}
    legend_toggle={false}
    loader={
      <div>
        Carregando Gráfico
      </div>
    }
    options={
      Object {
        "backgroundColor": "#000000",
        "curveType": "function",
        "hAxis": Object {
          "textStyle": Object {
            "color": "#FFFFFF",
          },
        },
        "legend": Object {
          "textStyle": Object {
            "color": "white",
          },
        },
        "vAxis": Object {
          "gridlines": Object {
            "color": "transparent",
          },
          "textPosition": "none",
        },
      }
    }
    render={null}
    rootProps={
      Object {
        "data-testid": "2",
      }
    }
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="500px"
  >
    <ContextProvider
      value={
        Object {
          "chartActions": null,
          "chartEvents": null,
          "chartType": "LineChart",
          "chartWrapperParams": Object {},
          "className": "",
          "columns": null,
          "controls": null,
          "data": Array [
            Array [
              "ano",
              "Usuários",
            ],
            Array [
              "2016",
              134,
            ],
            Array [
              "2017",
              523,
            ],
            Array [
              "2018",
              400,
            ],
            Array [
              "2019",
              843,
            ],
            Array [
              "2020",
              1432,
            ],
          ],
          "diffdata": null,
          "formatters": null,
          "getChartEditor": null,
          "getChartWrapper": [Function],
          "graphID": null,
          "graph_id": null,
          "height": "300px",
          "legendToggle": false,
          "legend_toggle": false,
          "loader": <div>
            Carregando Gráfico
          </div>,
          "options": Object {
            "backgroundColor": "#000000",
            "curveType": "function",
            "hAxis": Object {
              "textStyle": Object {
                "color": "#FFFFFF",
              },
            },
            "legend": Object {
              "textStyle": Object {
                "color": "white",
              },
            },
            "vAxis": Object {
              "gridlines": Object {
                "color": "transparent",
              },
              "textPosition": "none",
            },
          },
          "render": null,
          "rootProps": Object {
            "data-testid": "2",
          },
          "rows": null,
          "spreadSheetQueryParameters": Object {
            "gid": 1,
            "headers": 1,
          },
          "spreadSheetUrl": null,
          "style": Object {},
          "toolbarID": null,
          "toolbarItems": null,
          "width": "500px",
        }
      }
    >
      <div>
        Carregando Gráfico
      </div>
      <GoogleChartLoader
        onError={[Function]}
        onLoad={[Function]}
      >
        <Script
          attributes={Object {}}
          onCreate={[Function]}
          onError={[Function]}
          onLoad={[Function]}
          url="https://www.gstatic.com/charts/loader.js"
        />
      </GoogleChartLoader>
    </ContextProvider>
  </Chart>
</CurveLineChart>
`;
